x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go

```diff
@@ -189,14 +192,49 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 
 		for namespace, namespaceDetails := range namespaceDetailTotal {
 			m.logger.Debugf("Collected metrics from namespace %s", namespace)
-
 			// filter listMetricsOutput by detailed configuration per each namespace
 			filteredMetricWithStatsTotal := filterListMetricsOutput(listMetricsOutput, namespace, namespaceDetails)
 
 			// get resource type filters and tags filters for each namespace
 			resourceTypeTagFilters := constructTagsFilters(namespaceDetails)
 
-			eventsWithIdentifier, err := m.createEvents(svcCloudwatch, svcResourceAPI, filteredMetricWithStatsTotal, resourceTypeTagFilters, regionName, startTime, endTime)
+			//Check whether namespace is APIGW
+			checkns := "AWS/ApiGateway"
+			checkresource_type := "apigateway:restapis"
+			useonlyrest := false
+			if len(resourceTypeTagFilters) == 1 {
+				for key := range resourceTypeTagFilters {
+					if strings.Compare(strings.ToLower(key), strings.ToLower(checkresource_type)) == 0 {
+						useonlyrest = true
+					}
+				}
+			}
+			if strings.Contains(strings.ToLower(namespace), strings.ToLower(checkns)) {
+				// inforestapi includes only Rest APIs
+				if useonlyrest {
+					infoapi, err = aws.GetAPIGatewayRestAPIOutput(svcRestAPI, config.LimitRestAPI)
+					if err != nil {
+						m.Logger().Errorf("could not get rest apis output: %v", err)
+					}
+				} else {
+					infoapi, err = aws.GetAPIGatewayRestAPIOutput(svcRestAPI, config.LimitRestAPI)
+					if err != nil {
+						m.Logger().Errorf("could not get rest apis output: %v", err)
+					}
+					// infoapi includes only WebSocket and HTTP APIs
+					infotherapi, err := aws.GetAPIGatewayAPIOutput(svcHttpAPI)
+					if err != nil {
+						m.Logger().Errorf("could not get http and websocket apis output: %v", err)
+					}
+					if len(infotherapi) > 0 {
+						maps.Copy(infoapi, infotherapi)
```

Comment:
There is this lint error which I think can be ignored. 
The alternative can be sth like:
```
func mergeMaps(m1 map[string]string, m2 map[string]string) {
	for k, v := range m2 {
		m1[k] = v
	}
}
```

WDYT?

Reply:
seems this is the first time `maps.Copy` used in beats codebase :D that said, I think it could be due to an outdated linter. I think ignore should be fine